### PR TITLE
DM-15695 final acceptance summary statistics

### PR DESCRIPTION
**Pull request by parejkoj**

### CODE REVIEW

File: bin/summarizePerformanceRst.py [hunk 1]
```diff
@@ -66,6 +68,8 @@ def plotMetricScatter(df, name1, name2, band,
         Name of y-axis metric.
     band : `str`
         Filter band to plot.
+    descriptions : `dict` of `str`
+        Descriptions of the names, to inform the plot axes.
```
Comment (TallJimbo) [inline bin/summarizePerformanceRst.py]: FWIW, even with this added I still wouldn't be able to guess how to populate this dict.

File: bin/summarizePerformanceRst.py [hunk 2]
```diff
@@ -123,68 +127,130 @@ def plotMetricScatter(df, name1, name2, band,
...
+def main():
+    import argparse
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("-p", "--plot", action="store_true",
+                        help="Generate metric comparison plots.")
+    parser.add_argument("-v", "--verbose", action="store_true",
+                        help="Be more verbose when reading and computing statistics.")
+    parser.add_argument("-i", "--interactive", action="store_true",
+                        help="Open an ipdb console before exiting.")
+    args = parser.parse_args()
```
Comment (TallJimbo) [inline bin/summarizePerformanceRst.py]: Probably not worth changing here, but I prefer putting the argument parsing down in `__main__` and unpacking the results into kwargs in `main()`, so you can call from an interactive Python prompt or notebook if you want to.  Just a pattern to consider.

File: bin/summarizePerformanceRst.py [hunk 3]
```diff
@@ -123,68 +127,130 @@ def plotMetricScatter(df, name1, name2, band,
...
+    def print_y_is_less(x, y, name, verbose=False):
+        """Print a green `>` if y is less than x, otherwise a red `<`."""
+        if (verbose or x < y):
+            print(name, x, "\033[92m>\033[0m" if x > y else "\033[91m<\033[0m", y)
+
+    def print_y_is_less2(x, N, sigma, y, name, verbose=False):
+        """Print a green `>` if y is less than x+N*sigma, otherwise a red `<`."""
+        threshold = x + N*sigma
+        if (verbose or threshold < y):
+            less = "\033[92m>\033[0m" if threshold > y else "\033[91m<\033[0m"
```
Comment (TallJimbo) [inline bin/summarizePerformanceRst.py]: Color codes seem like a good place to use f-strings for clarity.

**Review by TallJimbo — APPROVED** [review]
Calculations look correct.  I've made some completely gratuitous style comments.

File: bin/summarizePerformanceRst.py (hunk 1 above)
Comment (parejkoj) [reply bin/summarizePerformanceRst.py]: Well, I think you wrote the original dict as a global, so I made a guess at what it was.

File: bin/summarizePerformanceRst.py (hunk 2 above)
Comment (parejkoj) [reply bin/summarizePerformanceRst.py]: This version is my "why would anyone ever call this except from the commandline?" version. I personally prefer the:

```
if __name__ == "__main__":
    import sys
    main(sys.argv[1:])
```

style, keeping the argparse in `main()` (or even lifting it to a `parse()` method of its own).

File: bin/summarizePerformanceRst.py (hunk 3 above)
Comment (parejkoj) [reply bin/summarizePerformanceRst.py]: Hmm... I think I came up with a neat solution there. Hadn't really used f-strings before, but I can see the utility.

**Pull request merged**